src/botPage/view/View.js

```diff
@@ -467,8 +468,16 @@ export default class View {
         });
     }
     stop() {
+        try {
+            const { contractId } = this.blockly.interpreter.bot.tradeEngine;
+            if (contractId) {
+                api.subscribeToOpenContract(contractId);
+            }
+            addStopListeners();
```

Comment:
This extra function doesn't add any value either regarding usability or readability, better to replace it with the instructions inside.